lstchain/image/muon/intensity_fitter.py

```diff
@@ -0,0 +1,562 @@
+"""
+Class for performing a HESS style 2D fit of muon images
+
+To do:
+    - Deal with astropy untis better, currently stripped and no checks made
+    - unit tests
+    - create container class for output
+
+"""
+import numpy as np
+from scipy.ndimage.filters import correlate1d
+from iminuit import Minuit
+from astropy import units as u
+from scipy.constants import alpha
+from scipy.stats import norm
+from astropy.coordinates import SkyCoord
+from functools import lru_cache
+
+from ctapipe.containers import MuonEfficiencyContainer
+from ctapipe.coordinates import CameraFrame, TelescopeFrame
+from ctapipe.core import TelescopeComponent
+from ctapipe.core.traits import FloatTelescopeParameter, IntTelescopeParameter
+
+
+# ratio of the areas of the unit circle and a square of side lengths 2
+CIRCLE_SQUARE_AREA_RATIO = np.pi / 4
+
+
+def chord_length(radius, rho, phi):
+    """
+    Function for integrating the length of a chord across a circle
+
+    Parameters
+    ----------
+    radius: float
+        radius of circle
+    rho: float
+        fractional distance of impact point from circle center
+    phi: ndarray in radians
+        rotation angles to calculate length
+
+    Returns
+    -------
+    ndarray: chord length
+    """
+    scalar = np.isscalar(phi)
+    phi = np.array(phi, ndmin=1, copy=False)
+
+    chord = 1 - (rho ** 2 * np.sin(phi) ** 2)
+    valid = chord >= 0
+
+    if rho <= 1.0:
+        # muon has hit the mirror
+        chord[valid] = radius * (np.sqrt(chord[valid]) + rho * np.cos(phi[valid]))
+    else:
+        # muon did not hit the mirror
+        chord[valid] = 2 * radius * np.sqrt(chord[valid])
+
+    chord[~valid] = 0
+
+    if scalar:
+        return chord[0]
+    return chord
+
+
+def intersect_circle(mirror_radius, r, angle, hole_radius=0):
+    """Perform line integration along a given axis in the mirror frame
+    given an impact point on the mirror
+
+    Parameters
+    ----------
+    angle: ndarray
+        Angle along which to integrate mirror
+
+    Returns
+    --------
+    float: length from impact point to mirror edge
+
+    """
+    mirror_length = chord_length(mirror_radius, (r / mirror_radius), angle)
+
+    if hole_radius == 0:
+        return mirror_length
+
+    hole_length = chord_length(hole_radius, (r / hole_radius), angle)
+    return mirror_length - hole_length
+
+
+def pixels_on_ring(radius, pixel_diameter):
+    """Calculate number of pixels of diameter ``pixel_diameter`` on the circumference
+    of a circle with radius ``radius``
+    """
+    circumference = 2 * np.pi * radius
+    n_pixels = u.Quantity(circumference / pixel_diameter)
+    return int(n_pixels.to_value(u.dimensionless_unscaled))
```

Comment:
```
n_pixels = circumference / pixel_diameter
return n_pixels
```
should do, no?

+ incomplete  docstring (do you expect astropy quantities here? - it will work in any case but better be explicit)

Reply:
This function works both with and without. It is actually not used I think in the current code.